man/sd_event_add_memory_pressure.xml

```diff
@@ -0,0 +1,268 @@
+
+  <refmeta>
+    <refentrytitle>sd_event_add_memory_pressure</refentrytitle>
+    <manvolnum>3</manvolnum>
+  </refmeta>
+
+  <refnamediv>
+    <refname>sd_event_add_memory_pressure</refname>
+    <refname>sd_event_source_set_memory_pressure_type</refname>
+    <refname>sd_event_source_set_memory_pressure_period</refname>
+    <refname>sd_event_trim_memory</refname>
+
+    <refpurpose>Add and configure an event source run as result of memory pressure</refpurpose>
+  </refnamediv>
+
+  <refsynopsisdiv>
+    <funcsynopsis>
+      <funcsynopsisinfo>#include &lt;systemd/sd-event.h&gt;</funcsynopsisinfo>
+
+      <funcsynopsisinfo><token>typedef</token> struct sd_event_source sd_event_source;</funcsynopsisinfo>
+
+      <funcprototype>
+        <funcdef>int <function>sd_event_add_memory_pressure</function></funcdef>
+        <paramdef>sd_event *<parameter>event</parameter></paramdef>
+        <paramdef>sd_event_source **<parameter>ret_source</parameter></paramdef>
+        <paramdef>sd_event_handler_t <parameter>handler</parameter></paramdef>
+        <paramdef>void *<parameter>userdata</parameter></paramdef>
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>int <function>sd_event_source_set_memory_pressure_type</function></funcdef>
+        <paramdef>sd_event_source *<parameter>source</parameter></paramdef>
+        <paramdef>const char *<parameter>type</parameter></paramdef>
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>int <function>sd_event_source_set_memory_pressure_period</function></funcdef>
+        <paramdef>sd_event_source *<parameter>source</parameter></paramdef>
+        <paramdef>uint64_t <parameter>threshold_usec</parameter></paramdef>
+        <paramdef>uint64_t <parameter>window_usec</parameter></paramdef>
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>int <function>sd_event_trim_memory</function></funcdef>
+        <paramdef>void</paramdef>
+      </funcprototype>
+    </funcsynopsis>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><function>sd_event_add_memory_pressure()</function> adds a new event source that is triggered
+    whenever memory pressure is seen. This functionality is built around the Linux kernel's <ulink
+    url="https://docs.kernel.org/accounting/psi.html">Pressure Stall Information (PSI)</ulink> logic.</para>
+
+    <para>Expects an event loop object as first parameter, and returns the allocated event source object in
+    the second parameter, on success. The <parameter>handler</parameter> parameter is a function to call when
+    memory pressure is seen, or <constant>NULL</constant>. The handler function will be passed the
+    <parameter>userdata</parameter> pointer, which may be chosen freely by the caller. The handler may return
+    negative to signal an error (see below), other return values are ignored. If
+    <parameter>handler</parameter> is <constant>NULL</constant>, a default handler that compacts allocation
```

Comment:
I think it would be good to document here that the default handler also logs structured data to the journal, otherwise it's a bit hard to find (without stumbling on it when looking at the journal or the catalogue)